### PR TITLE
[AF-1432] Remove AF dependencies from Uberfire Core Grids component

### PR DESCRIPTION
**Pull request by danielezonca**
Removed all UF/WB related dependencies to Uberfire Core Grids component to become a pure Lienzo/Gwt component and maybe moved to an independent Lienzo-Grid component in the future.

- Removed `PortablePreconditions` usages
- Changed `Callback` to `Consumer`
- Changed `Command` to `Runnable`
- Changed `Pair` to `Map.Entry`

@manstis @mdproctor @romartin @jsoltes 

PRs related:
https://github.com/kiegroup/appformer/pull/425
https://github.com/kiegroup/drools-wb/pull/902
https://github.com/kiegroup/kie-wb-common/pull/2009

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java [hunk 1]
```diff
@@ -348,9 +349,9 @@ protected void executeRenderQueueCommands(final boolean isSelectionLayer) {
         final List<Pair<Group, GridRenderer.RendererCommand>> gridLineCommands = new ArrayList<>();
         final List<Pair<Group, GridRenderer.RendererCommand>> allOtherCommands = new ArrayList<>();
         final List<Pair<Group, GridRenderer.RendererCommand>> selectedCellsCommands = new ArrayList<>();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java]: @danielezonca would you mind to convert also
```java
final List<Pair<Group, GridRenderer.RendererCommand>> gridLineCommands = new ArrayList<>();
final List<Pair<Group, GridRenderer.RendererCommand>> allOtherCommands = new ArrayList<>();
final List<Pair<Group, GridRenderer.RendererCommand>> selectedCellsCommands = new ArrayList<>();
```

to

```java
final List<Map.Entry<Group, GridRenderer.RendererCommand>> gridLineCommands = new ArrayList<>();
final List<Map.Entry<Group, GridRenderer.RendererCommand>> allOtherCommands = new ArrayList<>();
final List<Map.Entry<Group, GridRenderer.RendererCommand>> selectedCellsCommands = new ArrayList<>();
```

Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java]: Sorry probably PR description is a bit misleading (I did it for `appformer` module and then copy/paste in all the related PRs).
The scope of the PR is to update only UberFire Wires Core Grids component with those changes and then update all the corresponding usages. It is not intended to change all `Pair` with `Map.Entry` in the WB codebase.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java [hunk 2]
```diff
@@ -348,9 +349,9 @@ protected void executeRenderQueueCommands(final boolean isSelectionLayer) {
...
-        for (Pair<Group, List<GridRenderer.RendererCommand>> p : renderQueue) {
-            final Group parent = p.getK1();
-            final List<GridRenderer.RendererCommand> commands = p.getK2();
+        for (Map.Entry<Group, List<GridRenderer.RendererCommand>> p : renderQueue) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java]: @danielezonca thanks for repsonse, however I am still not sure why we need to stop use `Pair` as a iterator in cycle but is fine to store demanded items in `List<Pair>`, could you please explain? The used `Pair` is from package `org.uberfire.commons.data`. If it is fine, maybe this change is not needed at all? Not sure, just asking.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java]: It is slightly clearer, thanks.

Comment (danielezonca) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/widgets/grid/BaseExpressionGrid.java]: @jomarko yes I confirm @manstis 's comment. We are trying to isolate some components to become GWT/Lienzo only to be able to extract them from KIE (see also [this PR](https://github.com/kiegroup/kie-wb-common/pull/1977))
Then I agree with you that we can also decide to uniform some other KIE internal code that are using these libraries to the new interfaces but I would like to create another PR if needed.

**Pull request merged**